app/components/CreateAccountSlide/index.js

```diff
@@ -87,106 +92,142 @@ const SeedsContainer = styled.div`
-      >
-        Next
-      </ActionButton>
-    </Fragment>
-  )
-
-
-}
-const CreateAccount = ({importAddress}) => {
-  return (
-    <Fragment>
-      <div className='title'>
-        Your Tezos account seed is backed up!
-      </div>
-      <div className='description'>
-        Make sure to keep your seed phrase in a safe place. If you forget your seed phrase, you will not be able to recover your account. We do not store your seed phrase and cannot help you recover it if you lose it.
-      </div>
-
-      <ActionButton 
-        buttonTheme="primary"
-        onClick={importAddress}
-      >
-        Create Account
-      </ActionButton>
-    </Fragment>
-  )
-
-
-}
+        }
+        <div className="generate-part" onClick={this.updateMnemonic}>
+          <RefreshIcon
+            className='refresh-icon'
+            style={{ fill: '#2c7df7', transform: 'scaleX(-1)' }}
+          />
+          Generate Another Seed Pharse
+        </div>
+        <ActionButton
+          buttonTheme="primary"
+          onClick={() => this.nextAccountSlide(1)}
+        >
+          Next
+        </ActionButton>
+      </Fragment>
+    )
+  };
+
+  importAddress = () => {
+    const { seed } = this.state;
+    this.props.importAddress(GENERATE_MNEMONIC, seed);
+  };
+
+  createAccount = () => {
+    return (
+      <Fragment>
+        <div className='title'>
+          Your Tezos account seed is backed up!
+        </div>
+        <div className='description'>
+          Make sure to keep your seed phrase in a safe place. If you forget your seed phrase, you will not be able to recover your account. We do not store your seed phrase and cannot help you recover it if you lose it.
+        </div>
 
-export default function CreateAccountSlide(props: Props) {
-  const {currentSlide, seed, nextAccountSlide, importAddress, generateNewMnemonic} = props;
-  return(
-    <CreateAccountSlideContainer>
-      { !!currentSlide && 
-        <div className="back-part" onClick={() => nextAccountSlide(0)}>
+        <ActionButton
+          buttonTheme="primary"
+          onClick={importAddress}
+        >
+          Create Account
+        </ActionButton>
+      </Fragment>
+    )
+  };
+
+  render() {
+    const { currentSlide, seed } = this.state;
+    return(
+      <CreateAccountSlideContainer>
+        { !!currentSlide &&
+        <div className="back-part" onClick={() => this.nextAccountSlide(0)}>
           <ChevronLeftIcon
             className='chevron-icon'
             style={{ fill: '#4486f0' }}
           />
           Back to Seed Pharse
         </div>
-      }
-      {{
-          0: <ShowSeedPhrase seed={seed} nextAccountSlide={nextAccountSlide} generateNewMnemonic={generateNewMnemonic} />,
-          1: <BackUpSeedPhrase seed={seed} nextAccountSlide={nextAccountSlide} />,
-          2: <CreateAccount importAddress={importAddress} />,
-      }[currentSlide]}
-    </CreateAccountSlideContainer>
+        }
+        {
+          currentSlide === 0
```

Comment:
It might be more readable to but these three cases in a function and use a `switch` statement to dictate which gets rendered